dotCMS/src/integration-test/java/com/dotcms/publishing/PublisherAPITest.java

```diff
@@ -319,4 +326,293 @@ private FileFilter getNoBundleXMLFileFilter() {
+
+    private void createFilterDescriptor(final String key,final String title,final boolean defaultFilter,Map<String,Object> filtersMap){
+        if(!UtilMethods.isSet(filtersMap)){
+            final List<Object> listExcludeClasses = new ArrayList();
+            listExcludeClasses.add("User");
+            listExcludeClasses.add("OSGI");
+            filtersMap =
+                    ImmutableMap.of("dependencies",true,"relationships",true,"excludeClasses",
+                            listExcludeClasses);
+        }
+        publisherAPI.addFilterDescriptor(new FilterDescriptor(key,title,filtersMap,defaultFilter,"DOTCMS_BACK_END_USER"));
+    }
+
+    /**
+     * This test creates and gets a FilterDescriptor using the key as reference.
+     */
+    @Test
+    public void test_getFilterDescriptorByKey_success(){
+        final String key = "TestByKeySuccess.yml";
+        createFilterDescriptor(key,"TestByKeySuccess",true,null);
+        final FilterDescriptor filterDescriptor = publisherAPI.getFilterDescriptorByKey(key);
+        Assert.assertNotNull(filterDescriptor);
+        Assert.assertEquals(key,filterDescriptor.getKey());
+    }
+
+    /**
+     * This test creates 2 FilterDescriptors (one set as default) tries to get a FilterDescriptor, but since key is not passed,
+     * it returns the FilterDescriptor set as default.
+     */
+    @Test
+    public void test_getFilterDescriptorByKey_emptyKey_returnDefaultFilter(){
+        publisherAPI.getFilterDescriptorMap().clear();
+
+        final String keyDefault = "TestByKeyEmptyKeyDefault.yml";
+        createFilterDescriptor(keyDefault,"TestByKeyEmptyKeyDefault",true,null);
+
+        final String keyNonDefault = "TestByKeyEmptyKeyNonDefault.yml";
+        createFilterDescriptor(keyNonDefault,"TestByKeyEmptyKeyDefault",false,null);
+
+        final FilterDescriptor filterDescriptor = publisherAPI.getFilterDescriptorByKey("");
+        Assert.assertNotNull(filterDescriptor);
+        Assert.assertEquals(keyDefault,filterDescriptor.getKey());
+    }
+
+    /**
+     * This test creates 2 FilterDescriptors (one set as default) tries to get a FilterDescriptor, but since key passed does not belong to any FilterDescriptor,
+     * it returns the FilterDescriptor set as default.
+     */
+    @Test
+    public void test_getFilterDescriptorByKey_filterKeyDoesNotExist_returnDefaultFilter(){
+        publisherAPI.getFilterDescriptorMap().clear();
+
+        final String keyDefault = "TestByKeyDoesNotExistDefault.yml";
+        createFilterDescriptor(keyDefault,"TestByKeyDoesNotExistDefault",true,null);
+
+        final String keyNonDefault = "TestByKeyDoesNotExistNonDefault.yml";
+        createFilterDescriptor(keyNonDefault,"TestByKeyDoesNotExistNonDefault",false,null);
+
+        final FilterDescriptor filterDescriptor = publisherAPI.getFilterDescriptorByKey("thisKeyNotExists");
+        Assert.assertNotNull(filterDescriptor);
+        Assert.assertEquals(keyDefault,filterDescriptor.getKey());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)